torch/distributed/_functional_collectives.py

```diff
@@ -353,6 +353,20 @@ def all_to_all_single(
     return _maybe_wrap_tensor(tensor)
 
 
+def permute_tensor(self: torch.Tensor, src_dst_pairs: List[Tuple[int, int]], group: RANK_TYPES, tag: str = "") -> torch.Tensor:
+    world_size = dist.get_world_size()
+    output_split_sizes = [0] * world_size
+    input_split_sizes = [0] * world_size
+    for src, dst in src_dst_pairs:
+        if src == dist.get_rank():
```

Comment:
similarly, the src/dst rank should also be group rank instead of global rank iiuc